### PR TITLE
Promote `queryRewrite` over `SECURITY_CONTEXT`

### PR DESCRIPTION
**Pull request by hassankhan**
Story details: https://app.clubhouse.io/cube-dev/story/728

### CODE REVIEW

**Comment by anubhavsahoo** [comment]
Just came here to say.
absolutely ❤️ queryRewrite. Have been wishing for this feature for about a year now.
This greatly simplifies multi-tenant design. Good stuff!

File: docs/content/Cube.js-Backend/REST-API.md [hunk 1]
```diff
@@ -16,10 +16,13 @@ with **basePath** described in
 
 ### Authentication
 
-Cube.js uses API Token for requests' authorization and also for passing
-additional security context, which could be used in the
-[SECURITY_CONTEXT](/schema/reference/cube#context-variables-security-context)
-object in the Data Schema.
+Cube.js uses API tokens for request authorization and also for passing
```
Comment (rpaik) [inline docs/content/Cube.js-Backend/REST-API.md:19]:
```suggestion
Cube.js uses API tokens to request authorization and also for passing
```

File: docs/content/Auth/Security-Context.md [hunk 2]
```diff
@@ -28,12 +28,21 @@ the Data Schema and as the [`securityContext`][ref-config-sec-ctx] property
...
-[`SECURITY_CONTEXT`][ref-schema-sec-ctx] global variable in the Cube.js Data
-Schema.
+You can use [`queryRewrite`][ref-config-queryrewrite] to amend incoming queries
+with filters. For example, let's take the following query:
+
+```json
+{
+  "dimensions": ["Orders.status"],
+  "measures": ["Orders.count", "Orders.total"],
+  "timeDimension": "Orders.createdAt"
```
Comment (keydunov) [inline docs/content/Auth/Security-Context.md:40]: @hassankhan 

```js
  timeDimensions: [{
    dimension: 'Orders.createdAt',
    dateRange: ['2015-01-01', '2015-12-31'],
    granularity: 'month'
  }],
```

File: docs/content/Auth/Security-Context.md [hunk 3]
```diff
@@ -52,26 +61,27 @@ Schema.
...
-  )}`,
-
-  measures: {
-    count: {
-      type: `count`,
-    },
+module.exports = {
+  queryRewrite: (query, { securityContext }) => {
+    // Ensure `securityContext` has an `id` property
+    if (!securityContext.user_id) {
```
Comment (keydunov) [inline docs/content/Auth/Security-Context.md:71]: I don't think it is best practice to throw an error here. I'd rather return the query.

```js
if (!securityContext.user_id) {
  return query;
}
```

@paveltiunov @igorlukanin wdyt?

**Comment by keydunov** [comment]
Usage with Pre-Aggregations should into compile context section now because queryRewrite doesn't need it.

File: docs/content/Auth/Security-Context.md (hunk 3 above)
Comment (hassankhan) [reply docs/content/Auth/Security-Context.md:71]: @keydunov Is it worth also adding a `checkAuth()` property here to do the JWT verification separately, similar to the tests in https://github.com/cube-js/cube.js/pull/2568?

**Pull request merged**